api/router.go

```diff
@@ -76,7 +78,74 @@ func Serve() {
 	}
 }
 
-// GetMuxVars wraps github.com/gorilla/mux.Vars in order to mitigate context key issues in multi-repo projects.
-func GetMuxVars(r *http.Request) map[string]string {
-	return mux.Vars(r)
+type mainHandler struct {
+	mux *mux.Router
+}
+
+func (mh *mainHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	_ = module.RunWorker("http request", func(_ context.Context) error {
+		return mh.handle(w, r)
+	})
+}
+
+func (mh *mainHandler) handle(w http.ResponseWriter, r *http.Request) error {
+	// Setup context trace logging.
+	ctx, tracer := log.AddTracer(r.Context())
+	lrw := NewLoggingResponseWriter(w, r)
+	// Add request context.
+	apiRequest := &Request{
+		Request: r,
+	}
+	ctx = context.WithValue(ctx, requestContextKey, apiRequest)
+	// Add context back to request.
+	r = r.WithContext(ctx)
+
```

Comment:
It's better to use `r.Clone(ctx)`. See https://golang.org/pkg/net/http/#Request.WithContext